classes/04class/exercises/c04-iac02/fabioacp/vpc.tf

```diff
@@ -0,0 +1,185 @@
+locals {
+  nat_gateway_count = var.single_nat_gateway ? 1 : var.one_nat_gateway_per_az ? length(var.azs) : length(var.private_subnets)
+  nat_gateway_ips = split(",", join(",", aws_eip.nat.*.id))
+}
+
+################
+# VPC
+################
+resource "aws_vpc" "this" {
+  cidr_block           = var.vpc_cidr
+  instance_tenancy     = var.instance_tenancy
+  enable_dns_support   = var.dns_support
+  enable_dns_hostnames = var.dns_host_names
+
+  tags = merge(
+    var.tags,
+    {
+      "Name" = "${var.name}-iac"
+    },
+  )
+}
+
+################
+# Public subnet
+################
+resource "aws_subnet" "public" {
+  count = length(var.public_subnets) > 0 && length(var.public_subnets) >= length(var.azs) ? length(var.public_subnets) : 0
+
+  vpc_id                  = aws_vpc.this.id
+  cidr_block              = element(concat(var.public_subnets, [""]), count.index)
+  availability_zone       = length(regexall("^[a-z]{2}-", element(var.azs, count.index))) > 0 ? element(var.azs, count.index) : null
+  availability_zone_id    = length(regexall("^[a-z]{2}-", element(var.azs, count.index))) == 0 ? element(var.azs, count.index) : null
```

Comment:
using https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/availability_zones can certainly reduce this parsing overhead